### PR TITLE
Update API to json2csv version 4

### PR DESCRIPTION
**Pull request by eloyekunle**
v4 Changes that concern us:

- The actual JSON parson is now done via a `parse` method (i.e. From `json2csv(...)` to `json2csv.parse(...)`.
- The arguments accepted by the `parse` method is slightly different from that accepted by the previous `json2csv` function (From `json2csv({data, fields})` to `json2csv.parse(data, {fields})`.
- The parse is no longer done asynchronously, but the `parse` method generates and returns the CSV synchronously.

Fixes opencollective/opencollective#1777
Closes #1770

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "intl-messageformat": "2.2.0",
     "isomorphic-fetch": "2.2.1",
     "joi": "14.3.1",
-    "json2csv": "3.4.2",
+    "json2csv": "^4.3.5",
```
Comment (znarf) [inline package.json]: Small detail, we use fixed versions.

Comment (eloyekunle) [reply package.json]: Okay, fixed.

**Comment by znarf** [comment]
Thank you. Looks good!

There is currently a problem right now on CI that is not related to the work on this PR. Let's wait for the CI to be back on track.

**Comment by eloyekunle** [comment]
Okay great.

**Comment by znarf** [comment]
CI should be ok now. Can you rebase with master and update the branch?

**Comment by eloyekunle** [comment]
@znarf I've rebased with master.

File: cron/daily/charge_subscriptions.js [hunk 2]
```diff
@@ -61,26 +61,26 @@ async function run(options) {
...
-              {
-                filename: `${new Date().toLocaleDateString()}.csv`,
-                content: csv,
-              },
-            ];
-            await emailReport(start, orders, groupProcessedOrders(data), attachments);
-          }
+    vprint(options, 'Writing the output to a CSV file');
+    try {
+      const csv = json2csv({ data, fields: csvFields });
```
Comment (znarf) [inline cron/daily/charge_subscriptions.js]: Looks like the arguments are not matching the new API.

Comment (eloyekunle) [reply cron/daily/charge_subscriptions.js]: Okay, fixed.

**Comment by eloyekunle** [comment]
@znarf I've addressed the comment.

**Comment by znarf** [comment]
Thank you @eloyekunle! Merging.

A $100 bounty was attached to this issue: https://github.com/opencollective/opencollective/issues/1777.

To claim the bounty, you can:
 - a) submit an invoice to https://opencollective.com/engineering
 - b) or ask for an Open Collective gift card of the same value (to donate to other collectives)

**Comment by eloyekunle** [comment]
@znarf I've submitted an invoice on the Open Collective engineering website.

**Pull request merged**